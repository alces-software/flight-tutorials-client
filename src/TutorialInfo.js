// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

import React from 'react';
import type { TutorialType } from './types';

const TutorialInfo = ({
  tutorial,
} : {
  tutorial: TutorialType
}) => (
  <div>
    <h2>{tutorial.title}</h2>
    <div dangerouslySetInnerHTML={{ __html: tutorial.description }} />
  </div>
);

export default TutorialInfo;
