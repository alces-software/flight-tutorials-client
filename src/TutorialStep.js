import React, {Component} from 'react';

export default class TutorialStep extends Component {
  render() {
    const { current, completed, step } = this.props;
    if (!current && !completed) {
      return null;
    }
    return (
      <div>
        <h3>{step.title}</h3>
        <div dangerouslySetInnerHTML={{__html: step.description}} />
      </div>
    );
  }
}
