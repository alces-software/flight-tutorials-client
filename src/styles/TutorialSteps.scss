/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

.TutorialStep-title {
    a:hover {
        text-decoration: none;
        color: inherit;
        cursor: inherit;
    }
    .btn {
        margin-top: -4px;
    }
}
