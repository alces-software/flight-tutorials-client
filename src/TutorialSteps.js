// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

import React from 'react';

import type { StepMap }  from './types';
import TutorialStep from './TutorialStep';

const TutorialSteps = ({
  completedSteps,
  currentStep,
  steps,
} : {
  completedSteps: Array<string>,
    currentStep: string,
    steps: StepMap,
}) => (
  <div>
    {
      Object.keys(steps).map(stepName => (
        <TutorialStep
          key={stepName}
          completed={completedSteps.includes(stepName)}
          current={stepName === currentStep}
          step={steps[stepName]}
        />
      ))
    }
  </div>
);

export default TutorialSteps;
