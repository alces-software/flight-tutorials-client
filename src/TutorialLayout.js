// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

import React from 'react';
import { Grid, Row, Col, Button } from 'react-bootstrap';

import StandardModal from 'flight-common/lib/components/StandardModal';

import CloseButton from './CloseButton';
import TutorialInfo from './TutorialInfo';
import TutorialSteps from './TutorialSteps';
import StarRule from './StarRule';
import type { TutorialType } from './types';

type PropsType = {
  completedSteps : Array<string>,
  currentStep: string,
  onSessionRestartAccepted: () => void,
  onSessionRestartDeclined: () => void,
  onShowAllTutorials: () => void,
  terminal : React$Element<*>,  // A ReactTerminal element.
  tutorial: TutorialType,
  requestSessionRestart: boolean,
}

const TutorialLayout = ({
  completedSteps,
  currentStep,
  onSessionRestartAccepted,
  onSessionRestartDeclined,
  onShowAllTutorials,
  requestSessionRestart,
  terminal,
  tutorial,
} : PropsType) => (
  <div>
    <CloseButton onClose={onShowAllTutorials} />
    <div className="container">
      <h2 className="flight-tutorials-header">Flight Compute Tutorial</h2>
      <StarRule variant="primary" />
    </div>
    <Grid fluid >
      <Row>
        <Col xs={12} sm={12} md={4} lg={4} lgOffset={1} >
          <TutorialInfo tutorial={tutorial} />
          <TutorialSteps
            completedSteps={completedSteps}
            currentStep={currentStep}
            steps={tutorial.steps}
          />
        </Col>
        <Col xs={12} sm={12} md={8} lg={7} >
          <StandardModal
            buttons={
              <Button
                bsStyle="success"
                onClick={onSessionRestartAccepted}
              >
                Restart
              </Button>
            }
            show={requestSessionRestart}
            onHide={onSessionRestartDeclined}
            title="Your terminal session has been terminated"
          >
            Your terminal session has been terminated. Would you like to
            restart it?
          </StandardModal>
          {terminal}
        </Col>
      </Row>
    </Grid>
  </div>
);

export default TutorialLayout;
