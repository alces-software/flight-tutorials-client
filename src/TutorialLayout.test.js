// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TutorialInfo from './TutorialInfo';
import TutorialLayout from './TutorialLayout';
import TutorialStep from './TutorialStep'
import TutorialSteps from './TutorialSteps'

const steps = {
  step1: {
    title: 'My step 1',
    description: 'My step 1 description',
    matches: [],
  },
  step2: {
    title: 'My step 2',
    description: 'My step 2 description',
    matches: [],
  }
};

const tutorial = {
  title: 'My tutorial',
  description: 'My tutorial description',
  firstStep: 'step1',
  steps: steps,
};

const completedSteps = ['step1'];
const currentStep = 'step2';
const dummyTerminal = <div>dummy terminal</div>;

const renderTutorialLayout = () => (
  <TutorialLayout
    completedSteps={completedSteps}
    currentStep={currentStep}
    onShowAllTutorials={() => {}}
    showAllTutorialsButton
    terminal={dummyTerminal}
    tutorial={tutorial}
  />
);

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(renderTutorialLayout(), node);
});

it('renders correctly', () => {
  const tree = renderer.create(
    renderTutorialLayout()
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('includes an overview of the tutorial', () => {
  const wrapper = shallow(renderTutorialLayout());

  expect(wrapper).toContainReact(<TutorialInfo tutorial={tutorial} />);
});

it('includes the tutorial steps', () => {
  const wrapper = shallow(renderTutorialLayout());

  expect(wrapper).toContainReact(<TutorialSteps
    completedSteps={completedSteps}
    currentStep={currentStep}
    steps={steps}
  />);
});

it('includes the terminal', () => {
  const wrapper = shallow(renderTutorialLayout());

  expect(wrapper).toContainReact(dummyTerminal);
});
