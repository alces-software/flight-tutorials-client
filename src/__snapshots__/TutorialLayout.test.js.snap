// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="flight-CloseButton btn btn-link"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="lr"
    >
      <div
        className="rl"
      />
    </div>
  </button>
  <div
    className="container"
  >
    <h2
      className="flight-tutorials-header"
    >
      Flight Compute Tutorial
    </h2>
    <hr
      className="star-primary"
    />
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-lg-offset-1 col-md-4 col-sm-12 col-xs-12"
      >
        <div>
          <h2>
            My tutorial
          </h2>
          <div>
            <p>
              My tutorial description
            </p>
          </div>
        </div>
        <div
          className="panel-group"
          role="tablist"
        >
          <div
            className="panel panel-success"
            id={null}
          >
            <div
              className="panel-heading"
            >
              <h4
                className="panel-title"
                role="presentation"
              >
                <a
                  aria-controls={undefined}
                  aria-expanded={false}
                  aria-selected={false}
                  className="collapsed"
                  href={undefined}
                  onClick={[Function]}
                  role="tab"
                >
                  Step 1 My step 1
                </a>
              </h4>
            </div>
            <div
              aria-expanded={null}
              aria-hidden={true}
              className="panel-collapse collapse"
              id={undefined}
              role="tabpanel"
            >
              <div
                className="panel-body"
              >
                <div>
                  <p>
                    My step 1 description
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="panel panel-primary"
            id={null}
          >
            <div
              className="panel-heading"
            >
              <h4
                className="panel-title"
                role="presentation"
              >
                <a
                  aria-controls={undefined}
                  aria-expanded={true}
                  aria-selected={true}
                  className={null}
                  href={undefined}
                  onClick={[Function]}
                  role="tab"
                >
                  Step 2 My step 2
                </a>
              </h4>
            </div>
            <div
              aria-expanded={null}
              aria-hidden={false}
              className="panel-collapse collapse in"
              id={undefined}
              role="tabpanel"
            >
              <div
                className="panel-body"
              >
                <div>
                  <p>
                    My step 2 description
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-7 col-md-8 col-sm-12 col-xs-12"
      >
        <div />
        <div>
          dummy terminal
        </div>
      </div>
    </div>
  </div>
</div>
`;
